packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/DeleteRulesTests.java

```diff
@@ -0,0 +1,182 @@
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.rest.rules;
+
+import static java.util.stream.Collectors.toList;
+
+import static org.alfresco.utility.constants.UserRole.SiteCollaborator;
+import static org.alfresco.utility.constants.UserRole.SiteManager;
+import static org.alfresco.utility.report.log.Step.STEP;
+import static org.junit.Assert.assertTrue;
+import static org.springframework.http.HttpStatus.FORBIDDEN;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.http.HttpStatus.NO_CONTENT;
+import static org.springframework.http.HttpStatus.OK;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.alfresco.rest.RestTest;
+import org.alfresco.rest.model.RestRuleModel;
+import org.alfresco.rest.model.RestRuleModelsCollection;
+import org.alfresco.utility.model.FolderModel;
+import org.alfresco.utility.model.SiteModel;
+import org.alfresco.utility.model.TestGroup;
+import org.alfresco.utility.model.UserModel;
+import org.springframework.http.HttpStatus;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * Tests for DELETE /nodes/{nodeId}/rule-sets/{ruleSetId}/rules/{ruleId}.
+ */
+@Test(groups = {TestGroup.RULES})
+public class DeleteRulesTests extends RestTest
+{
+    private static final String FAKE_NODE_REF = "fake-node-id";
+
+    private UserModel user;
+    private SiteModel site;
+    private FolderModel ruleFolder;
+    private List<RestRuleModel> createdRules;
+    private RestRuleModel createdRuleA;
+
+    @BeforeClass(alwaysRun = true)
+    public void dataPreparation()
+    {
+        STEP("Create a user, site and folder");
+        user = dataUser.createRandomTestUser();
+        site = dataSite.usingUser(user).createPublicRandomSite();
+        ruleFolder = dataContent.usingUser(user).usingSite(site).createFolder();
+
+        STEP("Create rules in the folder");
+        createdRules = Stream.of("ruleA", "ruleB")
+                .map(ruleName -> {
+                    RestRuleModel ruleModel = new RestRuleModel();
+                    ruleModel.setName(ruleName);
+                    return restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingDefaultRuleSet()
+                            .createSingleRule(ruleModel);
+                })
+                .collect(toList());
+        createdRuleA = createdRules.get(0);
+    }
+
+    /**
+     * Delete a rule by its id.
+     */
+    @Test(groups = {TestGroup.REST_API, TestGroup.RULES, TestGroup.SANITY})
+    public void deleteSingleRuleAndGet204()
+    {
+        STEP("Attempt delete the rule ");
+        restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingDefaultRuleSet().deleteRule(createdRuleA.getId());
```

Comment:
I don't think this is a good idea. The test results will depend on the order of the tests.  If we create a second test that tries to delete the same rule then I assume both will pass, even though the second one didn't really delete anything.

Reply:
I agree with @tpage-alfresco, e2es should be independent of each other and be able to run on its own.

Reply:
Thanks - fixed that - it is independent.

Reply:
I don't see how it is independent.  The rule is created in a BeforeClass?

Reply:
I pushed the changes after the comment.